pkg/verifier/safe_delete.go

```diff
@@ -68,20 +72,29 @@ func BackupAndDelete(ctx context.Context, logger log.Logger, bkt, backupBkt objs
 	}
 
 	// Block uploaded, so we are ok to remove from src bucket.
-	level.Info(logger).Log("msg", "Deleting block", "id", id.String())
-	if err := block.Delete(ctx, logger, bkt, id); err != nil {
-		return errors.Wrap(err, "delete from source")
+	if deleteDelay.Seconds() == 0 {
+		level.Info(logger).Log("msg", "Deleting block", "id", id.String())
+		if err := block.Delete(ctx, logger, bkt, id); err != nil {
+			return errors.Wrap(err, "delete from source")
+		}
+	} else {
```

Comment:
Again, we can remove `else` here and simplify flow here! (:

Reply:
Actually you did that below ;p

Reply:
if and else automatically comes into my mind, so it was there. I have removed it now.